scripts/jest/ts-preprocessor.js

```diff
@@ -71,18 +71,17 @@ function compile(content, contentFilename) {
     getNewLine: function() {
       return '\n';
     },
+    fileExists: function (filename) {
+      return ts.sys.fileExists(filename);
+    },
   };
   var program = ts.createProgram([
     'lib.d.ts',
     'jest.d.ts',
     contentFilename,
   ], tsOptions, compilerHost);
-  var errors = program.getDiagnostics();
-  if (!errors.length) {
-    var checker = program.getTypeChecker(true);
-    errors = checker.getDiagnostics();
-    checker.emitFiles();
-  }
+  var emitResult = program.emit();
+  var errors = ts.getPreEmitDiagnostics(program).concat(emitResult.diagnostics);
```

Comment:
As far as I remember, the change is that now TypeScript doesn't throw errors, but rather collects & returns them to you from those methods. This way, you can do whatever you want on host side - either output all of them to stderr, or throw the first one if not empty or just check if there were any and so on.